src/openzeppelin/tests/token/test_dual20.cairo

```diff
@@ -0,0 +1,354 @@
+use array::ArrayTrait;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::testing::set_contract_address;
+use traits::Into;
+
+use openzeppelin::tests::mocks::camel20_mock::CamelERC20Mock;
+use openzeppelin::tests::mocks::erc20_panic::SnakeERC20Panic;
+use openzeppelin::tests::mocks::erc20_panic::CamelERC20Panic;
+use openzeppelin::tests::mocks::non721_mock::NonERC721;
+use openzeppelin::tests::mocks::snake20_mock::SnakeERC20Mock;
+use openzeppelin::token::erc20::dual20::DualERC20;
+use openzeppelin::token::erc20::dual20::DualERC20Trait;
+use openzeppelin::token::erc20::interface::IERC20CamelDispatcher;
+use openzeppelin::token::erc20::interface::IERC20CamelDispatcherTrait;
+use openzeppelin::token::erc20::interface::IERC20Dispatcher;
+use openzeppelin::token::erc20::interface::IERC20DispatcherTrait;
+use openzeppelin::tests::utils;
+
+///
+/// Constants
+///
+
+const NAME: felt252 = 111;
+const SYMBOL: felt252 = 222;
+const DECIMALS: u8 = 18_u8;
+
+fn SUPPLY() -> u256 {
+    2000_u256
+}
+fn VALUE() -> u256 {
+    300_u256
+}
+fn OWNER() -> ContractAddress {
+    contract_address_const::<10>()
+}
+fn SPENDER() -> ContractAddress {
+    contract_address_const::<20>()
+}
+fn RECIPIENT() -> ContractAddress {
+    contract_address_const::<30>()
+}
+fn OPERATOR() -> ContractAddress {
+    contract_address_const::<40>()
+}
+
+///
+/// Setup
+///
+
+fn setup_snake() -> (DualERC20, IERC20Dispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(SUPPLY().low.into());
+    calldata.append(SUPPLY().high.into());
+    calldata.append(OWNER().into());
+    let target = utils::deploy(SnakeERC20Mock::TEST_CLASS_HASH, calldata);
+    (DualERC20 { contract_address: target }, IERC20Dispatcher { contract_address: target })
+}
```

Comment:
We can do this to avoid serializing everything manually:

```suggestion
use openzeppelin::tests::utils::SerializedAppend;

fn setup_snake() -> (DualERC20, IERC20Dispatcher) {
    let mut calldata = ArrayTrait::new();
    calldata.append_serde(NAME);
    calldata.append_serde(SYMBOL);
    calldata.append_serde(SUPPLY());
    calldata.append_serde(OWNER());
    let target = utils::deploy(SnakeERC20Mock::TEST_CLASS_HASH, calldata);
    (DualERC20 { contract_address: target }, IERC20Dispatcher { contract_address: target })
}
```

Having this in utils:

```
trait SerializedAppend<T> {
    fn append_serde(ref self: Array<felt252>, value: T);
}

impl SerializedAppendImpl<T, impl TSerde: Serde<T>, impl TDrop: Drop<T>> of SerializedAppend<T> {
    fn append_serde(ref self: Array<felt252>, value: T) {
        value.serialize(ref self);
    }
}
```

Reply:
Haha good call! I've been serializing like this as of recently:
```
let mut calldata: Array<felt252> = ArrayTrait::new();
Serde::serialize(@var1, ref calldata);
Serde::serialize(@var2, ref calldata);
```
but your solution is much better. Thank you, sir!